utilities/ordered-set.metta

```diff
@@ -0,0 +1,88 @@
+;; data structure for exemplars -- aka scored trees -- Xmplr
+;; deme ID
+(: demeID Type)
+(: mkDemeID (-> Number demeID))
+
+;; composite_score
+(: cscore Type)
+(: getCscore (-> tree cscore))
+        
+;; composite score components
+(: scoreT Type)
+(: cpxyT Type)
+
+(: mkScoreT (-> Number scoreT))
+(: mkCpxy (-> Number cpxyT))
+
+;; composite score is composed of a raw score, comlexity score, a complexity penalty and a uniformity penalty score
+(: mkCscore (-> scoreT cpxyT scoreT scoreT cscore))
+
+;; behavioral_score
+(: bscore Type)
+(: mkBscore (-> Number bscore))
+
+(: Xmplr Type)
+
+(: mkXmplr (-> tree demeID cscore bscore Xmplr))
+
+;; XXX - SUGGESTION -- for the exemplars, I have tried to follow the type definitions as they are defined in C++ MOSES
+;; although now I don't exactly see the advantage using specific types for numerical values such as bscore, score ..
+;; I say it is better better to uses these as Numbers without wrapper type definitions
+
+;; data structures for the metapopulation
+;; keeping the same naming convention as used in the definitio of the ordered multiset
+;; Ordered set data structure -- OS
+
+(: OS (-> $a Type))
+(: mkOS (-> $a (OS $a)))
```

Comment:
Also I think the type definition should be a recursive one like list. Refer eyobed's multiset definition in dev branch to see what I mean.

Reply:
Also, this file should contain information only about the ordered set since the file name is `ordered-set.metta`. The other logic should be in a different file like `mosesTypes.metta` for example.